src/main/java/it/gov/pagopa/idpay/transactions/controller/TransactionsControllerImpl.java

```diff
@@ -0,0 +1,33 @@
+package it.gov.pagopa.idpay.transactions.controller;
+
+import it.gov.pagopa.idpay.transactions.exception.ClientExceptionWithBody;
+import it.gov.pagopa.idpay.transactions.model.RewardTransaction;
+import it.gov.pagopa.idpay.transactions.service.RewardTransactionService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Flux;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+@RestController
+@Slf4j
+public class TransactionsControllerImpl implements TransactionsController{
+    private final RewardTransactionService rewardTransactionService;
+
+    public TransactionsControllerImpl(RewardTransactionService rewardTransactionService) {
+        this.rewardTransactionService = rewardTransactionService;
+    }
+
+    @Override
+    public Flux<RewardTransaction> findAll(String idTrxIssuer, String userId, LocalDateTime trxDateStart, LocalDateTime trxDateEnd, BigDecimal amount) {
+        if(idTrxIssuer != null){
+            return rewardTransactionService.findByIdTrxIssuer(idTrxIssuer,userId,trxDateStart, trxDateEnd, amount);
+        }else if(userId != null && trxDateStart != null && trxDateEnd != null){
+            return rewardTransactionService.findByUserIdAndRangeDateAndAmount(userId, trxDateStart, trxDateEnd, amount);
```

Comment:
let's keep generic the name of this method (eg: findByRange)

otherwise we should update it for each new required filter :D